feat(sms) : #15 - 전화번호 인증 기능 구현

kathyleesh opened the pull request:
<!-- Please check the one that applies to this PR using "x". -->
## 어떤 이유로 MR를 하셨나요?
- [x] feature 병합
-  버그 수정
-  코드 개선
-  기타(아래에 자세한 내용을 기입해주세요)
<!-- 
feat : 새로운 기능을 추가
fix : 버그 수정 또는 기능에 대한 큰 변화와 결과에 변화가 있을 때
docs : 문서 관련 커밋
refactor : 기능에 대한 변화 없이 리팩토링
style : 코드 스타일 변경(formatting, missing semi colons, …)
test : 테스트 관련 커밋
chore : 기타 커밋
-->

## 세부 내용 - 왜 해당 MR이 필요한지 자세하게 설명
- 인증번호 전송
- 인증번호 확인

## MR하기 전에 확인해주세요
- [x] Merge 제목 확인
- [x] Approve 1인 이상
- [x] Merge 위치 확인 (develop)

## Git Issue Number
<!-- #000 번호만 입력 -->
- #15

kathyleesh committed:
feat(build.gradle) : #15 - SMS와 redis 추가

net.nurigo, redis gradle 추가

Ref: #1
Related to: #15

kathyleesh committed:
feat(SmsProperties) : #15 - coolSMS와 연결

환경변수 들고와서 coolSMS와 연결

Ref: #2
Related to: #15

kathyleesh committed:
feat(SmsService) : #15 - 인증 메세지 내용 생성

- makeSmsContent : 메세지 생성
- makeRandomNumber : 무작위 랜덤 숫자 추출
- builderCertificationContent : 메세지 틀

Ref: #2
Related to: #15

kathyleesh committed:
feat(SmsService) : #15 - 인증 메세지 발송 및 저장 기능 구현

coolSms API를 이용하여 인증번호 발송하고, 발송 정보를 Redis에 저장

Ref: #2
Related to: #15

kathyleesh committed:
feat(SmsService) : #15 - 인증 메세지 확인 기능 구현

사용자가 입력한 인증번호가 Redis에 저장된 인증번호와 동일한지 확인

Ref: #2
Related to: #15

kathyleesh committed:
feat(SmsDao) : #15 - 인증번호 redis 저장 기능 구현

- 사용자가 입력한 휴대폰 번호와 인증번호를 저장하고 TTL을 3분으로 설정
- Redis에 해당 휴대폰번호(KEY)로 저장된 인증번호(VALUE)가 존재하는지 확인

Ref: #2
Related to: #15

kathyleesh committed:
feat(SmsController) : #15 - 전화번호 인증 api 구현

- sendSms : 인증번호 발송
- SmsVerification : 인증번호 확인

Ref: #2
Related to: #15

syhan7516 commented on backend/authentication-service/src/main/java/com/authenticationservice/sms/SmsController.java:
```diff
@@ -0,0 +1,37 @@
+public class SmsController {
+    private final SmsService smsService;
+
+    // 인증번호 발송
+    @PostMapping("/sms/sends")
+    public ResponseEntity<? extends BaseResponseBody> sendSms(@RequestBody SmsReqDto requestDto) {
+        System.out.println(requestDto.getPhone());
+        smsService.sendSms(requestDto.getPhone());
+        return ResponseEntity.status(HttpStatus.CREATED).body(BaseResponseBody.of(0,"Success"));
+    }
```
@RequestBody SmsReqDto requestDto 부분에서
String phone만 데이터가 필요한데 requestDto 말고 따로 만들어서 사용하는건 어떨까요?

kathyleesh replied:
<img width="843" alt="스크린샷 2023-09-06 오전 11 57 27" src="https://github.com/SOL-Child/SOL-Family/assets/90851865/82963cdb-e4b2-4e30-8841-6c8cc596570b">
수정완료하였습니다!

syhan7516 commented:
수고하셨습니다!

kathyleesh committed:
refactor(SmsSendReqDto) : #15 - SmsSendReqDto로 추가 및 RequestBody 수정

- 인증번호 전송 요청 시 phone 데이터만 보낼 수 있도록 SmsSendReqDto 생성
- SmsController RequestBody 수정

Ref: #1
Related to: #15
